src/screens/TopicPageScreen/MemoizedListComponent.js

```diff
@@ -171,35 +218,35 @@ const RenderListFeed = (props) => {
             showScoreButton={showScoreButton}
           />
         </View>
-        {getCommentLength(item.latest_reactions.comment) > 0 ? (
-          <View style={styles.contentReaction(getHeightReaction())}>
-            <PreviewComment
-              user={item.latest_reactions.comment[0].user}
-              comment={item?.latest_reactions?.comment[0]?.data?.text || ''}
-              image={item?.latest_reactions?.comment[0]?.user?.data?.profile_pic_url || ''}
-              time={item.latest_reactions.comment[0].created_at}
-              totalComment={getCommentLength(item.latest_reactions.comment) - 1}
-              onPress={onPressComment}
-              item={item.latest_reactions.comment[0]}
-            />
-            <Gap height={8} />
-          </View>
-        ) : (
-          <TouchableOpacity
-            onPress={() => onPressComment(isHaveSeeMore)}
-            style={styles.contentReaction(getHeightReaction())}>
-            <WriteComment
-              postId={''}
+        <BlurredLayer
+          layerOnly
+          blurType="light"
+          withToast={true}
+          isVisible={isBlurred}
+          containerStyle={{
+            height: commentHeight()
+          }}>
+          {hasComment ? (
+            <View style={styles.contentReaction(getHeightReaction())}>
+              <PreviewComment
+                user={item.latest_reactions.comment[0].user}
+                comment={item?.latest_reactions?.comment[0]?.data?.text || ''}
+                image={item?.latest_reactions?.comment[0]?.user?.data?.profile_pic_url || ''}
+                time={item.latest_reactions.comment[0].created_at}
+                totalComment={getTotalReaction(item) - 1}
+                item={item.latest_reactions.comment[0]}
+                onPress={onPressComment}
+              />
+            </View>
+          ) : (
+            <AddCommentPreview
               username={handleUserName(item)}
-              value={''}
-              onChangeText={() => {}}
-              onPress={() => {}}
-              loadingPost={false}
-              isViewOnly={true}
-              withAnonymityLabel={false}
+              isBlurred={isBlurred}
+              heightReaction={getHeightReaction()}
+              onPressComment={() => onPressComment(isHaveSeeMore)}
             />
-          </TouchableOpacity>
-        )}
+          )}
+        </BlurredLayer>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `BlurredLayer` component has been introduced, which wraps either a `PreviewComment` or an `AddCommentPreview` component based on whether there are comments. Ensure that all the props passed to these components are handled correctly.
```javascript
<BlurredLayer
  layerOnly
  blurType="light"
  withToast={true}
  isVisible={isBlurred}
  containerStyle={{
    height: commentHeight()
  }}>
  {hasComment ? (
    <PreviewComment
      user={item.latest_reactions.comment[0].user}
      comment={item?.latest_reactions?.comment[0]?.data?.text || ''}
      image={item?.latest_reactions?.comment[0]?.user?.data?.profile_pic_url || ''}
      time={item.latest_reactions.comment[0].created_at}
      totalComment={getTotalReaction(item) - 1}
      item={item.latest_reactions.comment[0]}
      onPress={onPressComment}
    />
  ) : (
    <AddCommentPreview
      username={handleUserName(item)}
      isBlurred={isBlurred}
      heightReaction={getHeightReaction()}
      onPressComment={() => onPressComment(isHaveSeeMore)}
    />
  )}
</BlurredLayer>
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->